### ISSUE TITLE
lr or loss为0

### ISSUE THREAD

**Issue by newwindhxx** [cocoshe/MIMIGenRec#3]
您好，向您请教问题解决方法
在使用单卡a100微调训练时，只修改了:
### train
per_device_train_batch_size: 64
gradient_accumulation_steps: 32

sh文件里使用:
FORCE_TORCHRUN=1 llamafactory-cli train examples/train_full/Industrial_and_Scientific/industry_rec_full_sft_0.5b_dsz0.yaml

遇到要么loss一直为0，要么lr和grad_norm为0
这是部分训练日志：
{'loss': 12.1034, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.15}

  2%|▏         | 6/390 [01:33<1:26:53, 13.58s/it]
  2%|▏         | 7/390 [01:46<1:24:37, 13.26s/it]
                                                 
{'loss': 12.0582, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.18}

  2%|▏         | 7/390 [01:46<1:24:37, 13.26s/it]
  2%|▏         | 8/390 [01:58<1:23:06, 13.05s/it]
                                                 
{'loss': 12.0021, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.21}

  2%|▏         | 8/390 [01:58<1:23:06, 13.05s/it]
  2%|▏         | 9/390 [02:11<1:22:09, 12.94s/it]
                                                 
{'loss': 12.0653, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.23}

**Comment by cocoshe** [comment]
您好，这个数据集的 sft 完整 log 已经在 wandb 开源，见：https://wandb.ai/cocoshe/MIMIGenRec-public/runs/a0nwvtjg?nw=nwusercocoshe

<img width="2310" height="696" alt="Image" src="https://github.com/user-attachments/assets/c02b7eda-0370-4497-b14d-ae7677225a25" />

从日志来看，lr会像配置的那样，20个warmup，后做余弦decay

提供一些排查思路：
1. 改成原本配置的超参，是否能正常训练
2. 相同环境下，跑一下llamafactory的官方demo是否能正常（clone一个llamafactory的仓库，保持环境相同，跑他的demo）：https://llamafactory.readthedocs.io/zh-cn/latest/getting_started/sft.html

**Comment by cocoshe** [comment]
您好想问下这个问题解决了嘛？有卡点可以多多交流，如果解决了的话我就关闭这个issue了~

**Comment by newwindhxx** [comment]
> 您好想问下这个问题解决了嘛？有卡点可以多多交流，如果解决了的话我就关闭这个issue了~

谢谢您的耐心回答，问题已经解决了
